File: custom_addons/anar_woocommerce/static/src/js/notifications.js
/** @odoo-module **/


import { registry } from "@web/core/registry";
import { woocommerceNotificationChannelName } from "../config";


const productNotification = {

    dependencies: ['bus_service', 'notification'],

    start(env, { bus_service, notification }) {

        bus_service.addChannel(woocommerceNotificationChannelName)

        bus_service.addEventListener('notification', ({ detail: notifications }) => {

            if (notifications[0].type === 'woocommerce_notification') {

                const notificationPayload = notifications[0].payload
                const notificationTime = notification.add(
                    notificationPayload.message,
                    {
                        title: notificationPayload.title,
                        type: notificationPayload.type,
                        sticky: notificationPayload.sticky
                    }
                )

                if (!notificationPayload.sticky)
                    setTimeout(notificationTime, notificationPayload.timeout)

            }

        });

        bus_service.start();
    }

};


registry.category("services").add("productNotification", productNotification);

File: custom_addons/anar_woocommerce/utils/notifications.py
from odoo import _
from odoo.http import request
from ..conf import WOOCOMMERCE_NOTIFICATION_TIMEOUT


class Notification:

    def __init__(self, title, message, type, sticky):

        self.title = title
        self.message = message
        self.type = type
        self.sticky = sticky

        if not self.sticky:
            self.timeout = WOOCOMMERCE_NOTIFICATION_TIMEOUT

    def send_to_bus(self, channel_name):
        request.env['bus.bus']._sendone(channel=channel_name,
                                        message=self.__dict__,
                                        notification_type='woocommerce_notification')
